src/compiler/scala/tools/nsc/symtab/classfile/ClassfileParser.scala

```diff
@@ -549,7 +549,8 @@ abstract class ClassfileParser(reader: ReusableInstance[ReusableDataReader]) {
       val needsConstructor = (
            !sawPrivateConstructor
         && !(instanceScope containsName nme.CONSTRUCTOR)
-        && ((sflags & INTERFACE) == 0 || (sflags | JAVA_ANNOTATION) != 0)
+        && ((sflags & INTERFACE) == 0 || (sflags & JAVA_ANNOTATION) != 0)
```

Comment:
this certainly looks correct, so far every java interface had an `init` method.

```
scala> typeOf[Runnable].members
val res0: $r.intp.global.Scope =
Scope{
  def <init>(): Runnable;
```

It was added in https://github.com/scala/scala/pull/6869/files#diff-48440af466bc60ce77eb8afa75128abf7863f2bcba7a3153cf3adad0271df0b1

Reply:
sorry I missed this comment. I'll read the linked PR.